engine/engine.go

```diff
@@ -15,39 +16,42 @@ import (
 	"github.com/thrasher-corp/gocryptotrader/currency"
 	"github.com/thrasher-corp/gocryptotrader/currency/coinmarketcap"
 	"github.com/thrasher-corp/gocryptotrader/dispatch"
+	exchange "github.com/thrasher-corp/gocryptotrader/exchanges"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/asset"
 	"github.com/thrasher-corp/gocryptotrader/exchanges/request"
 	"github.com/thrasher-corp/gocryptotrader/exchanges/trade"
 	gctscript "github.com/thrasher-corp/gocryptotrader/gctscript/vm"
 	gctlog "github.com/thrasher-corp/gocryptotrader/log"
-	"github.com/thrasher-corp/gocryptotrader/portfolio"
 	"github.com/thrasher-corp/gocryptotrader/portfolio/withdraw"
 	"github.com/thrasher-corp/gocryptotrader/utils"
 )
 
-// Engine contains configuration, portfolio, exchange & ticker data and is the
+// Engine contains configuration, portfolio manager, exchange & ticker data and is the
 // overarching type across this code base.
 type Engine struct {
-	Config                      *config.Config
-	Portfolio                   *portfolio.Base
-	ExchangeCurrencyPairManager *ExchangeCurrencyPairSyncer
-	NTPManager                  ntpManager
-	ConnectionManager           connectionManager
-	DatabaseManager             databaseManager
-	GctScriptManager            *gctscript.GctScriptManager
-	OrderManager                orderManager
-	PortfolioManager            portfolioManager
-	CommsManager                commsManager
-	exchangeManager             exchangeManager
-	DepositAddressManager       *DepositAddressManager
-	Settings                    Settings
-	Uptime                      time.Time
-	ServicesWG                  sync.WaitGroup
+	Config                  *config.Config
+	apiServer               *apiServerManager
+	CommunicationsManager   *CommunicationManager
+	connectionManager       *connectionManager
+	currencyPairSyncer      *syncManager
+	DatabaseManager         *DatabaseConnectionManager
+	DepositAddressManager   *DepositAddressManager
+	eventManager            *eventManager
+	ExchangeManager         *ExchangeManager
+	ntpManager              *ntpManager
+	OrderManager            *OrderManager
+	portfolioManager        *portfolioManager
+	gctScriptManager        *gctscript.GctScriptManager
+	websocketRoutineManager *websocketRoutineManager
+	WithdrawManager         *WithdrawManager
+	Settings                Settings
+	uptime                  time.Time
+	ServicesWG              sync.WaitGroup
 }
 
-// Vars for engine
-var (
-	Bot *Engine
-)
+// Bot is a happy global engine to allow various areas of the application
+// to access its setup services and functions
+var Bot *Engine
```

Comment:
I just checked and we are much closer than I thought we were! The main piece of work to achieve this now is gctscript. I think it should be its own PR for that, rather than increase scope here

Reply:
👍 Will resolve